apiservice/apiservice.go

```diff
@@ -0,0 +1,240 @@
+package apiservice
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/grpcutils"
+)
+
+func NewFlowAPIService(accessNodeAddressAndPort flow.IdentityList, timeout time.Duration) (*FlowAPIService, error) {
+	accessClients := make([]access.AccessAPIClient, accessNodeAddressAndPort.Count())
+	for i, identity := range accessNodeAddressAndPort {
+		if identity.NetworkPubKey == nil {
+			clientRPCConnection, err := grpc.Dial(
+				identity.Address,
+				grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+				grpc.WithInsecure(), //nolint:staticcheck
+				backend.WithClientUnaryInterceptor(timeout))
+			if err != nil {
+				return nil, err
+			}
```

Comment:
do we abort init on a single access node being unreachable?

Reply:
My assessment is that users will require multiple upstream access nodes, if they use them to satisfy an overall bandwidth requirement.
If it is for performance, there should be a threshold, like 95% of capacity available, to be meaningful.
I suggest keeping a 100% startup ratio here, requiring reconnects by gRPC. We can negotiate a separate command line option that sets the startup threshold for the number of available upstream nodes.
If it is for reliability, and they cannot boot with three nodes, they should not boot with two, missing the reliability criteria.